modules/level-4/cm-1005-introduction-to-programming-i/README.md

```diff
@@ -105,5 +106,42 @@ _"Specific readings for each topic are listed with direct links to free online r
+- [ESLint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) - checks for "[code lint](https://en.wikipedia.org/wiki/Lint_(software))" in JS. Requires [ESLint](https://eslint.org/) to be installed, and may require some configuration to conform with the style used in the course (for more information on using ESLint, see the [ESLint user guide](https://eslint.org/docs/user-guide/configuring/); for a sample configuration that's compatible with ITP1, see [here](https://gist.githubusercontent.com/morags/2d762fd51c5ea4d733756baaf20cc6cc/raw/4093a82e966b38a75ba32bc0f623d61412ee047c/.eslintrc.json))
+- [Project Manager](https://marketplace.visualstudio.com/items?itemName=alefragnani.project-manager) - useful when you start working on several projects at a time
+
+To enable autocompletion for p5.js, first install [Node.js](https://nodejs.org/en/). Node is a JS [runtime](https://en.wikipedia.org/wiki/Runtime_system) - a program that runs JS code. It's not entirely unlike your browser's JS engine - in fact, it is based on Chrome's [V8 engine](https://en.wikipedia.org/wiki/V8_(JavaScript_engine)) - but it's intended for servers and commandline programs rather than web pages. Its package manager, [npm](https://www.npmjs.com/), is useful for setting up JS projects of any scale, so it's a good tool to have if you're going to work with JS. When you install Node, make sure to choose the option "add to PATH", so you have Node and npm available throughout your system. Once Node and npm are installed, open Code and a commandline interface (`cmd.exe` or `powershell.exe` in Windows, `Terminal` in macOS), navigate to your project folder (eg. `Documents/Goldsmiths/ITP1/Game_project`) and follow these steps:
+
+1. Type the following command into your commandline: `npm i --save-dev @types/p5` . This will tell npm to download type definitions for p5.js. You will notice that npm has created a `node_modules` folder and a couple of files, which it uses to track and manage downloaded packages; you needn't worry about those at the moment (for more information on npm, see the [npm Docs](https://docs.npmjs.com/cli/v7/commands/npm)).
+2. In Code, create a new file with the following text, and save it to the project folder as `jsconfig.json`:
+
+```json
+{
+  "compilerOptions": {},
+  "exclude": ["node_modules"],
+}
+```
+
+This will let Code know that you're working on a JS project and that it should expect some other configuration files like `globals.d.ts` to be present (for more configuration options, see the [jsconfig.json reference](https://code.visualstudio.com/docs/languages/jsconfig)).
+
+3. Create another file with this text, and save it to the project folder as `globals.d.ts`:
+
+```ts
+import {  } from "./node_modules/@types/p5/global";
```

Comment:
IMO all of this is way too complicated for someone just starting out. 

--- 


https://marketplace.visualstudio.com/items?itemName=samplavigne.p5-vscode

This extension seems to do all that's required to get functioning p5js completions.